enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupCommandBuilder.java

```diff
@@ -0,0 +1,358 @@
+                .toString();
+        return String.format( "--backup-dir=%s", location );
+    }
+
+    private String argBackupName( File targetLocation )
+    {
+        String backupName = Optional.ofNullable( targetLocation )
+                .map( File::getName )
+                .orElseThrow( wrongArguments( "No target location specified" ) );
+        return String.format( "--name=%s", backupName );
+    }
+
+    private static Supplier<IllegalArgumentException> wrongArguments( String message )
+    {
+        return () -> new IllegalArgumentException( message );
+    }
+
+    private String argFallbackToFull()
+    {
+        return Optional.ofNullable( fallbackToFull )
+                .map( flag -> String.format( "--fallback-to-full=%s", flag ) )
+                .orElse( "" );
+    }
+
+    private String argTimeout()
+    {
+        return Optional.ofNullable( this.timeout )
+                .map( value -> String.format("--timeout=%dms", value) )
+                .orElse( "" );
+    }
+
+    private String argCcOwners()
+    {
+        return Optional.ofNullable( this.consistencyCheckOwners )
+                .map( value -> String.format( "--check-consistency=%b", this.consistencyCheckOwners ) )
+                .orElse( "" );
+    }
+
+    private String argCcLabel()
+    {
+        return Optional.ofNullable( this.consistencyCheckLabel )
+                .map( value -> String.format( "--cc-label-scan-store=%b", this.consistencyCheckLabel ) )
+                .orElse( "" );
+    }
+
+    private String argCcIndexes()
+    {
+        return Optional.ofNullable( this.consistencyCheckIndexes )
+                .map( value -> String.format( "--cc-indexes=%b", this.consistencyCheckIndexes ) )
+                .orElse( "" );
+    }
+
+    private String argCcGraph()
+    {
+        return Optional.ofNullable( this.consistencyCheckGraph )
+                .map( value -> String.format( "--cc-graph=%b", this.consistencyCheckGraph ) )
+                .orElse( "" );
+    }
+
+    private String argAdditionalConf( File backupTarget )
+    {
+        if ( additionalConfig == null )
+        {
+            return "";
+        }
+        File configFile = neo4jHomeFromTarget( backupTarget ).resolve( "../additional_neo4j.conf" ).toFile();
+        writeConfigToFile( additionalConfig, configFile );
+
+        return String.format( "--additional-config=%s", configFile );
+    }
+
+    private void writeConfigToFile( Config config, File file )
+    {
+        try
+        {
+            Writer fileWriter = new BufferedWriter( new FileWriter( file ) );
+            for ( Map.Entry<String,String> entry : config.getRaw().entrySet() )
+            {
+                fileWriter.write( String.format( "%s=%s\n", entry.getKey(), entry.getValue() ) );
+            }
+            fileWriter.close();
+        }
+        catch ( IOException e )
+        {
+            e.printStackTrace();
```

Comment:
Does not look like a good handling of exception? should we be able to continue if this fails?

Reply:
It's part of a builder, so it shouldn't be throwing a checked exception but having it wrapped as unchecked should be fine. Simply printing out the trace is wrong, I admit I overlooked that.